lib/bamboo/adapters/send_grid_adapter.ex

```diff
@@ -19,7 +19,11 @@ defmodule Bamboo.SendGridAdapter do
       # In config/config.exs, or config.prod.exs, etc.
       config :my_app, MyApp.Mailer,
         adapter: Bamboo.SendGridAdapter,
-        api_key: "my_api_key" # or {:system, "SENDGRID_API_KEY"},
+        api_key: "my_api_key"
+          # or {:system, "SENDGRID_API_KEY"},
+          # or {ModuleName, :method_name, []}
+          # or fn -> "a resolveable function" end
+          # or &ModuleName.method_name/0
```

Comment:
AFAIK it is not possible to prepare a release with anonymous functions in config

Reply:
they have worked for me in the past, but that was using distillery, I haven't tested it with mix release. I can remove that suggestion too if you think it warrants it

Reply:
updated the docs to remove it, since it's definitely not a normal pattern and 

https://elixirforum.com/t/how-to-use-anonymous-functions-in-configuration-files/23069
https://elixirforum.com/t/anonymous-functions-in-config-why-does-it-work/22202